### PR TITLE
fix-3592: remove file schema from uri before loading it

### PR DESCRIPTION
**Pull request by dufoli**
tested on my side and fix the issue for me.

### CODE REVIEW

**Comment by gsmet** [comment]
@dmlloyd does it make sense to you?

**Comment by gsmet** [comment]
@dufoli what value do you have exactly?

I'm also concerned that we might not have a `!` every time and the code is not entirely safe.

**Comment by dmlloyd** [comment]
Hmm, something is fishy.  Dropping `file:` makes the URI relative (even if it starts with `/`), which forces it to be resolved against... `file:` on the next line, which should result in the same URI being given back again!

I'd like to see a test, or example, or some evidence of what this change actually does.

File: core/devmode/src/main/java/io/quarkus/dev/ClassLoaderCompiler.java [hunk 1]
```diff
@@ -104,6 +104,10 @@ public ClassLoaderCompiler(ClassLoader classLoader,
                                 for (String i : classPath.toString().split(" ")) {
                                     File f;
                                     try {
+                                        if (i.startsWith("file:")) {
+                                            i = i.substring(5, i.lastIndexOf('!'));
```
Comment (jaikiran) [inline core/devmode/src/main/java/io/quarkus/dev/ClassLoaderCompiler.java]: Hello @dufoli , won't this cause a problem is the `!` character isn't part of the `String` `i`, in which case `lastIndexOf` will return -1?

**Comment by dufoli** [comment]
meaculpa, it do not work at all.. I was failing on another error and because build is not stable, I was thinking this issue is fixed but no. Fix produce other error.
Can someonce close the PR? because I do not find how to on my side?

What I have was a stacktrace with ":" not allowed after file. So I was guessing that path must not been uri but regular local path. So I try to remove the file part.
I try few thing to remove file but never succeed to get absolute path working.
file:\C:\.... do not worked.
The odd thing to me is that we have file:\C:\... and not file:\\C:\... 

Anyway, I try with file:\\C:\, \C:\..., C:\,... it never worked.
I have an even before assert because the jar file is not found in temp forlder...

**Comment by jaikiran** [comment]
I have been able to narrow this down. A lengthy explanation will follow in the original issue. I think we'll need @dmlloyd's help to fix this properly.

**Comment by jaikiran** [comment]
https://github.com/quarkusio/quarkus/pull/4527 which has been merged, should take care of this.

**Comment by gsmet** [comment]
Yup, thanks for noticing, I will close this one.

**Pull request closed without merging**